Formula/minio.rb

```diff
@@ -21,12 +19,23 @@ def install
     clipath = buildpath/"src/github.com/minio/minio"
     clipath.install Dir["*"]
 
-    cd buildpath/"src/github.com/minio/minio/" do
-      system "go", "build", "-o", bin/"minio", "-v"
+    cd clipath do
+      if build.head?
+        system "go", "build", "-o", buildpath/"minio"
+      else
+        minio_release = `git tag --points-at HEAD`.chomp
+        minio_version = minio_release.gsub(/RELEASE\./, "").chomp
+        minio_commit = `git rev-parse HEAD`.chomp
+
+        system "go", "build", "-ldflags", "-X main.minioVersion=#{minio_version} -X main.minioReleaseTag=#{minio_release} -X main.minioCommitID=#{minio_commit}", "-o", buildpath/"minio"
+      end
     end
+
+    bin.install buildpath/"minio"
   end
 
   test do
-    system "#{bin}/minio"
+    minio_version = DateTime.parse(version).strftime("%Y-%m-%dT%H-%M-%SZ")
```

Comment:
Unfortunately 1.8 (which we still have to support for now) doesn't support `.parse` here so better maybe to just use the old, crappy test: https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/5998/version=mavericks/testReport/junit/brew-test-bot/mavericks/test_minio/